### ISSUE TITLE
Custom Blank Crashes Rack When Loading Images

### ISSUE THREAD

**Issue by McGroggin** [freddyz/computerscare-vcv-modules#62]
What are the requirements for acceptable images for the Custom Blank Module? Are there limitations on bit-depth, file size or any other factors?

I just tried loading 9 different images found online and locally, a mixture of random PNGs, GIFs and JPGs and only one loaded successfully. Every other attempt caused Rack to crash to desktop.

**Comment by McGroggin** [comment]
From log.txt:

[0.651 info src/window.cpp:72] Loaded SVG C:\Users\rober\DOCUME~1/Rack/plugins-v1/CountModula/res/Raven/Palette.svg
[1.816 debug src/ComputerscareBlank.cpp:895] path not module path
[1.816 debug src/ComputerscareBlank.cpp:896] path: empty, modulePath:C:\Users\rober\Downloads\etienne-jacob.gif
[2.036 fatal src/main.cpp:45] Fatal signal 22. Stack trace:
26: ZN4rack10appDestroyEv 0x456435
25: raise 0x8d17abe0
24: abort 0x8d17f1e0
23: ZN9__gnu_cxx12__to_xstringINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEcEET_PFiPT0_yPKS8_PcEySB_z 0x108ee920
22: ZN10SmoothKnobD1Ev 0x10889880
21: ZSt7shuffleIN9__gnu_cxx17__normal_iteratorIPiSt6vectorIiSaIiEEEESt26linear_congruential_engineIjLj16807ELj0ELj2147483647EEEvT_S9_OT0_ 0x1094a960
20: ZZN9__gnu_cxx6__stoaIffcJEEET0_PFT_PKT1_PPS3_DpT2_EPKcS5_PyS9_EN11_Save_errnoD1Ev 0x10952e00
19: ZNSt8_Rb_treeIiSt4pairIKiP12ParamAndTypeESt10_Select1stIS4_ESt4lessIiESaIS4_EE8_M_eraseEPSt13_Rb_tree_nodeIS4_E 0x10945eb0
18: ZN18ComputerscareBlank14setFrameDelaysESt6vectorIfSaIfEE 0x10895010
17: ZN11tPNGDisplayIN4rack6widget17TransparentWidgetEE4drawERKNS1_6Widget8DrawArgsE 0x1088a9b0
16: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
15: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
14: ZN4rack3app12ModuleWidget4drawERKNS_6widget6Widget8DrawArgsE 0x492bba
13: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
12: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
11: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
10: ZN4rack6widget10ZoomWidget4drawERKNS0_6Widget8DrawArgsE 0x4a83e2
9: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
8: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
7: ZN4rack2ui12ScrollWidget4drawERKNS_6widget6Widget8DrawArgsE 0x4a3fd4
6: ZN4rack6widget6Widget4drawERKNS1_8DrawArgsE 0x4a776c
5: ZN4rack6Window3runEv 0x4594ea
4: main 0x781840
3: main 0x781840
2: main 0x781840
1: BaseThreadInitThunk 0x8e157020
0: RtlUserThreadStart 0x8ea3d220

**Comment by freddyz** [comment]
Thanks for the report.  Honestly I don't have any idea what sorts of images will work and what will not. I've run into GIFs that don't load, but the only insight into "why" is some error message produced by Rack's image loading library.  I did not write any of the code that actually loads or displays images so I'm not sure how much more support I can offer, BUT if Rack is crashing I'd guess that means a bug in my module.  Would you be willing to share any of the actual image files that cause Rack to crash? I think you can upload them here along with a post.

**Comment by McGroggin** [comment]
Fair enough.

I gathered a new round of images to try. I had a greater success rate this time but still a number of crashes.

I saw a screenshot somewhere of the Blank module displaying a 'bad image' message. The only time I was presented with this was when restarting VCV AFTER a crash.

![moduleRestart](https://user-images.githubusercontent.com/18359151/109686269-1da6c180-7b50-11eb-924b-b806507018cd.PNG)

 If this worked more consistently the process of trial and error would be less daunting. Mostly I have to restart once or twice and accept clearing the patch to continue.

I can work with this as it is when it comes to found images. My only concern is that images I had created myself in Photoshop or with the Windows Snipping Tool also crash rack. I've tried exporting as PNG, JPG and BMP all without success. This kills the dream of creating my own custom blank panels for use in rack.

When images do load, they look and run great and I love the addition of the expander allowing for a level of control.

Test images that fail:

![three](https://user-images.githubusercontent.com/18359151/109686360-33b48200-7b50-11eb-851e-757cd193a614.jpg)
![one](https://user-images.githubusercontent.com/18359151/109686414-3fa04400-7b50-11eb-984b-dabbb2d38ae7.jpg)
![two](https://user-images.githubusercontent.com/18359151/109686437-45962500-7b50-11eb-8e10-9efb0b7c2abf.png)
![Untitled-1](https://user-images.githubusercontent.com/18359151/109686501-56469b00-7b50-11eb-96f6-ed59140722ba.jpg)
![GibsonVCV](https://user-images.githubusercontent.com/18359151/109686525-5cd51280-7b50-11eb-9bcb-cb62470a7122.png)

Out of the 20 images tested today, 9 loaded without issue.

**Comment by freddyz** [comment]
Hmm so I was able to load all of these with no problems running on my Mac.  Makes me wonder if perhaps it is a platform-specific bug like with file paths or something.  But that wouldn't quite make sense because you said some images do work....  I will try these out on my windows computer later and see if I can reproduce the problem.
<img width="1189" alt="Screen Shot 2021-03-02 at 11 21 31 AM" src="https://user-images.githubusercontent.com/1319733/109688443-e03f3580-7b49-11eb-98c0-80d066f0e60d.png">

**Comment by freddyz** [comment]
Could you upload the GIF that failed in your log.txt, `etienne-jacob.gif` ?  The stack trace there looks like some other bug in my code.

Also, when you have a moment if you could reproduce the crash when attempting to load one of these "problem images", and copy/paste the `log.txt` maybe there will be a clue as to what the problem is.

**Comment by adingGENESIS** [comment]
Hey there!

I'd like to chirp in to say that I'm also having the same problem, but if it helps, I am using a Windows system.
Out of all the images I've tried, all of them cause VCV rack to crash.

If you'd like, I can send the log in a later post in the thread as well as the images that don't work.

**Comment by freddyz** [comment]
Yes please do @adingGENESIS.  Just one image and the `log.txt` will probably suffice.  Was Custom Blank working properly for you prior to the recent 1.4 update?

**Comment by freddyz** [comment]
Alright I was able to reproduce the problem on my Windows computer.  Thanks a lot @McGroggin and @adingGENESIS for the information, I believe I have fixed the problem.  If you are able and willing to install the pre-release and check to make sure all your usual images load without crashing it would be much appreciated!  Windows binary can be found here for release candidate v1.4.1: https://github.com/freddyz/computerscare-vcv-modules/releases/tag/v1.4.1
